_posts/2024-08-06-quarkus-and-leyden.adoc

```diff
@@ -0,0 +1,275 @@
+---
+layout: post
+title: 'Project Leyden'
+date: 2024-08-06
+tags: leyden
+synopsis: 'Project Leyden: exploring its potential for Quarkus users.'
+author: leydenquarkus
+---
+:imagesdir: /assets/images/posts/leyden
+
+You might have heard of https://openjdk.org/projects/leyden/[Project Leyden]: an initiative within the OpenJDK project with very ambitious goals.
+
+When casually namedropping "I'm looking into Project Leyden" with some fellow Java developers these days, I would usually spot some off guard thinking: "Leyden..?", to then offer a hasty save "Ah, right, the project to improve startup times".
+
+Which is not wrong, as indeed startup times improvements are one of its goals, yet we believe the other stated goals of the project offer even greater potential for our favourite platform and its users.
+
+== What is Leyden?
+
+Project Leyden is an initiative from the OpenJDK team. It is an ongoing experiment that is currently being developed by the joint effort of teams from different companies.
+
+[quote]
+____
+The primary goal of this Project is to improve the startup time, time to peak performance, and footprint of Java programs.
+
+-- Project Leyden, first thing on its project page
+____
+
+Leyden is a general umbrella project to address slow startup and large footprint. To keep costs down, in all forms such as energy consumption, required hardware resources, and indeed money, it's useful to keep bootstrap times reasonably low, but it's even more effective to reduce the time to peak performance, such as the time it takes for the JVM to "warm up", and reducing the footprint of our applications, such as total memory, has very direct impact.
```

Comment:
I believe the sentence is too long and hard to parse, thus it would benefit a lot from a rewrite

Reply:
Leyden is a general umbrella project to address slow startup and large footprint. It is clearly useful to keep JDK bootstrap times and footprint low. That all helps to reduce energy consumption, hardware resource use and, ultimately, money. However, it's equally as important to reduce the time to application peak performance, time usually spent loading application classes and executing application code, including JIT compiling methods on hot code paths. Reducing application footprint can have a very big impact and this can be achieved by trimming not just application data but also application classes and code. Leyden is addressing ways that the JVM itself can help developers achieve those goals.

Reply:
It reads much better, thanks!

Reply:
applied, thanks :)